app/models/unit.rb

```diff
@@ -1634,7 +1634,8 @@ def import_task_files_from_zip(zip_file)
         if (File.extname(file.name) == '.pdf') || (File.extname(file.name) == '.zip')
           found = false
           task_definitions.each do |td|
-            next unless /^#{td.abbreviation}/ =~ file_name
+            # strip extensions and look for exact match
+            next unless td.abbreviation == File.basename(file.name, ".*")
```

Comment:
I dont want to force exact match - checking the start should be ok but we need to search for the longest match and use that.

Reply:
Done